### ISSUE TITLE
Support plans for DSTU-1

### ISSUE THREAD

**Issue by whbath** [jamesagnew/hapi-fhir#184]
Hi There,
In light of the near transition to DSTU-2 we are trying to work out what the plan should be for DSTU-1 support. Do you have plans for a timespan that DSTU-1 might be supported?

From what I've read in the documentation you can leverage DSTU1 Resource Providers to generate a DSTU-2 representation, can you equally use the DSTU-2 Resource Providers and expose a DSTU-1 representation off it?

In short, if we implemented the DSTU-2 providers, is providing a DSTU-1 representation more or a less a matter of [flipping the switch on the context object](http://jamesagnew.github.io/hapi-fhir/doc_dstu2.html) or is there more to it than that?

**Comment by jamesagnew** [comment]
[2015-06-09, 10:07:01 AM] James Agnew: Hi William- At this point we have no intent to drop support for DSTU1. It’s kind of hard to say what the long term holds in store. For the HL7 v2 library, we still support v2.1 (which is now roughly 20 years old) through 2.6, with support being added for 2.7 & 2.8 this year.
[2015-06-09, 10:09:34 AM] James Agnew: Ideally I’d like to do the same for FHIR, just supporting all versions indefinitely. I guess it’s will be interesting to see how FHIR develops though- There were more structural changes between DSTU1 & DSTU2 (e.g. the new bundle format) than there were between v2.1 and v2.6. At some point it may become too tedious to continue maintaining DSTU1 support, so I suppose we’d then need to decide what to do. Bottom line though, DSTU1 will be around for a while. We use it internally on projects we have in production, so we’re in no rush to be done with it.
[2015-06-09, 10:10:57 AM] James Agnew: In terms of what to target though, I really don’t know. Obviously all of the public discussion these days is around DSTU2, but DSTU2 is probably still months away from being finalized and it’s still changing. If you need stability now it
[2015-06-09, 10:12:33 AM] James Agnew: ’s probably best to target DSTU1. If you can handle a bit of churn and/or don’t intend to release until later this year, it’s probably best to target DSTU2. If you are want to really go nuts with it, you could always target both. Technically HAPI is fine with supporting both from a single JVM (as our public test server does). I’ve never tried this on a “real” application though.. Probably it would be a lot of work.
